logical_switch.go

```diff
@@ -331,6 +331,70 @@ func (odbi *ovndb) lsExtIdsDelImp(ls string, external_ids map[string]string) (*O
 	return &OvnCommand{operations, odbi, make([][]map[string]interface{}, len(operations))}, nil
 }
 
+// lsAuxAddImp is for adding records to external_ids or other_config tables
+func (odbi *ovndb) lsAuxAddImp(ls string, auxConf map[string]string, auxTable string) (*OvnCommand, error) {
+	if (auxTable != "external_ids") && (auxTable != "other_config") {
+		return nil, fmt.Errorf("%s is neither other_config or external_ids", auxTable)
+	}
+
+	var operations []libovsdb.Operation
+	row := make(OVNRow)
+	row["name"] = ls
+	lsuuid := odbi.getRowUUID(TableLogicalSwitch, row)
+	if len(lsuuid) == 0 {
+		return nil, ErrorNotFound
+	}
+	if len(auxConf) == 0 {
+		return nil, fmt.Errorf("%s map is nil or empty", auxTable)
+	}
+	mutateSet, err := libovsdb.NewOvsMap(auxConf)
+	if err != nil {
+		return nil, err
+	}
+	mutation := libovsdb.NewMutation(auxTable, opInsert, mutateSet)
```

Comment:
from RFC 7047, we have:

   <mutation>
      A 3-element JSON array of the form [<column>, <mutator>, <value>]
      
      map
         <mutator> must be "insert" or "delete".

         If <mutator> is "insert", then each of the key-value pairs in
         the map in <value> is added to <column> only if its key is not
         already present.  The required type of <value> is slightly
         relaxed, in that it may have fewer than the minimum number of
         elements specified by the column's type.

Say, if the external_ids column already has that key, then the call to this API will not update that key from the definition above.

So, should we first get the current external_ids, merge the to-be-set external_ids and then proceed? Am I missing something?

Reply:
if key is already present, then there's no other way of updating its value but to remove it and then insert this key back with new value, right?